### DISCUSSION TITLE
Propsal: Move docs from README to MkDocs

### DISCUSSION THREAD

**Discussion by Ra2-IFV** [Simple-Tracker/qBittorrent-ClientBlocker#136] [General]
Looks like the help docs are getting more complicated?
```markdown
[中文 (默认, Beta 版本)](https://github.com/Simple-Tracker/qBittorrent-ClientBlocker/blob/dev/README.md) [English (Default, Beta Version)](https://github.com/Simple-Tracker/qBittorrent-ClientBlocker/blob/dev/README.en.md)
[中文 (Public 正式版)](https://github.com/Simple-Tracker/qBittorrent-ClientBlocker/blob/master/README.md) [English (Public version)](https://github.com/Simple-Tracker/qBittorrent-ClientBlocker/blob/master/README.en.md)
```
I suggest you use MkDocs to manage documentations. [Demo](https://squidfunk.github.io/mkdocs-material/getting-started/) with Material theme

**Comment by Simple-Tracker** [comment]
Good idea. But the project is becoming very slow at this stage, so this plan may be postponed until the project can start to be active in the future.

**Reply by Ra2-IFV** [reply]
I can help, so I created this discussion to get your permission
So where do we start? Create a `docs` folder under this repository, or use a new repo?

**Reply by Simple-Tracker** [reply]
> I can help, so I created this discussion to get your permission
> So where do we start? Create a `docs` folder under this repository, or use a new repo?

If you can help, that would be great. I prefer to use an existing repository for convenience.

**Reply by Ra2-IFV** [reply]
Okay, I'll fork

**Reply by Ra2-IFV** [reply]
I need some help:

## Versioning

Do you need this? The docs can be versioned in two ways:

Use [mike](https://github.com/jimporter/mike), a plugin for mkdocs, versioning is managed by mkdocs.
Use git, and have different docs in different branches (e.g master and dev).
I prefer the first one, and if possible but may not necessary, store it in a different repo

## Multi language

[Material for MkDocs](https://github.com/squidfunk/mkdocs-material) has language selector widget, but it cannot change the nav bar because it's defined in the `mkdocs.yml` file:
![Screenshot_2024-11-13_003552](https://github.com/user-attachments/assets/8da4aee9-ca79-4841-b036-f9a0b2183b51)
![Screenshot_2024-11-13_003601](https://github.com/user-attachments/assets/33512b9b-711a-4c72-8297-6de334841071)
There is a [workaround](https://github.com/squidfunk/mkdocs-material/discussions/2346) for this.

**Reply by Simple-Tracker** [reply]
> I need some help:
> 
> ## Versioning
> Do you need this? The docs can be versioned in two ways:
> 
> Use [mike](https://github.com/jimporter/mike), a plugin for mkdocs, versioning is managed by mkdocs. Use git, and have different docs in different branches (e.g master and dev). I prefer the first one, and if possible but may not necessary, store it in a different repo
> 
> ## Multi language
> [Material for MkDocs](https://github.com/squidfunk/mkdocs-material) has language selector widget, but it cannot change the nav bar because it's defined in the `mkdocs.yml` file: ...

The documentation needs two versions: for the main branch (Release) and the dev branch (possibly beta).

Usually the commits of the main branch are only behind the dev branch, so if we can only maintain the documentation of the dev branch, it may reduce maintenance costs.

Regarding multilingualism, the navigation titles used by the current documentation include both languages, so it may not be necessary to use a workaround.
